v2v2 Build system

kristojorg opened the pull request:
This is a new version of the `chore/build-system` branch that is rebased on top of `v2v2` now. This is a draft PR for the moment until I verify everything is working. 🤞

kristojorg committed:
build file organization

kristojorg committed:
add build script and Node polyfills

kristojorg committed:
remove 'import *' from modules

This caused errors because it is no longer supported TS functionality that
was being back-filled by webpack. It's not necessary anyways though.

kristojorg committed:
move tsconfig to root

kristojorg committed:
add dev server

kristojorg committed:
add named 'build' export to index.ts

This is for interoperability with <script> usage, where otherwise
we would have had to use D2Reader.default.build. Now we can just
use `D2Reader.build`.

kristojorg committed:
replace reader.js with index.js in viewer

kristojorg committed:
clean up package.json scripts

kristojorg committed:
remove webpack

kristojorg committed:
remove unused packages

kristojorg committed:
add target browsers

kristojorg committed:
update fs/promises import

kristojorg committed:
remove unused files

kristojorg committed:
move D2Reader to reader.ts

kristojorg committed:
properly resolve types

kristojorg committed:
moves types to dist/types

kristojorg committed:
add 'script' export

kristojorg committed:
make UserSettings Partial in applyUserSettings

kristojorg committed:
build to separate folders

kristojorg committed:
init v2

kristojorg committed:
fix missing mediaoverlays setting

kristojorg committed:
remove unused vars and lint

kristojorg committed:
V2V2 New Streamer (#243)

* remove old streamer

* add server.ts

* clean up npm scripts

* move sample epubs

* update gitignore

* fix examples

* comments

* load local publications

* replace streamer

* fix esm error with ts-node

* unnecessary tsconfig

* remove unused var

* npm lint

* branch to demonstrate bug with setting initial appearance

* Remove custom CSS, fix bug in applying initial appearance setting.

* Remove index page demonstrating bug; fix bug instead.

* Bump patch version for bug fix.

* fix query selector bug

* Roll back package.json changes.

* minor fixes for injectables and clean

* no need for the default reader and fixed api reader

* fixes header and footer menu

* very minor ui fixes, and use of current settings.

* version bump to 1.9.1

Co-authored-by: Aferdita <aferdita.muriqi@gmail.com>
Co-authored-by: Alan Harnum <waharnum@gmail.com>

kristojorg committed:
lint fix

kristojorg committed:
ignore parcel folders when linting

kristojorg committed:
user dist/reader.js for iife bundle

kristojorg committed:
update chokidar

kristojorg committed:
remove console

aferditamuriqi commented on package.json:
```diff
@@ -14,26 +14,27 @@
+    "clean": "rimraf node_modules dist parcel-dist .parcel-cache",
+    "dev": "ts-node -T build.ts -w",
+    "build": "ts-node -T build.ts",
+    "streamed": "node \"./examples/streamed/server/server-cli.inlinesourcemap.js\" ./examples/streamed/epubs/",
```
do we still need this? since we got rid of the compiled server?

aferditamuriqi commented on viewer/index_dita.html:
```diff
@@ -503,6 +503,48 @@
+                        <button style="background-color: gainsboro;" id="dismissIcon"><i class="material-icons" style="color:gray">close</i></button>
+                    </div>
+                    <div id="highlight-toolbox-mode-add">
+                        <button style="display: unset;" id="colorIcon" class="color-option"><span></span></button>
+                        <button style="background-color: gainsboro;" id="highlightIcon"><span style="background: yellow;display: inline-block;width: 24px;height: 24px;">
+                                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" role="img" class="icon open" aria-labelledby="text-icon">
+                                    <title id="text-icon">Text</title>
+                                    <path d="M5 4v3h5.5v12h3V7H19V4z"></path>
+                                </svg>
+                            </span></button>
+                        <button style="background-color: gainsboro;" id="underlineIcon"><span style="display: inline-block;width: 24px;height: 24px;border-bottom: yellow solid 4px;">
+                                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" role="img" class="icon open" aria-labelledby="text-icon">
+                                    <title id="text-icon">Text</title>
+                                    <path d="M5 4v3h5.5v12h3V7H19V4z"></path>
+                                </svg>
+                            </span></button>
+                        <button style="background-color: gainsboro;" id="speakIcon"><span style="display: inline-block;width: 24px;height: 24px;">
+                                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" preserveAspectRatio="xMidYMid meet" role="img" class="icon open" aria-labelledby="speak-icon">
+                                    <title id="speak-icon">Speak</title>
+                                    <circle cx="9" cy="9" r="4"></circle>
+                                    <path
+                                      d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm7.76-9.64l-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27zM20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14z">
+                                    </path>
+                                    <path d="M0 0h24v24H0z" fill="none"></path>
+                                </svg>
+                            </span></button>
+                        <button style="background-color: gainsboro;" id="anIcon"><span style="display: inline-block;width: 24px;height: 24px;">
+                                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="black" width="24" height="24">
+                                    <path d="M0 0h24v24H0z" fill="none" />
+                                    <path
+                                      d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z" />
+                                </svg>
+                            </span></button>
+                    </div>
+                    <div id="highlight-toolbox-mode-edit">
+                        <button style="background-color: gainsboro;display: none" id="deleteIcon"><i class="material-icons" style="color:gray">delete</i></button>
+                    </div>
+                </span>
+        </div>
+
```
this block is duplicate. i can prepare a little pr with these findings. :)

aferditamuriqi committed:
removed streamed script

aferditamuriqi committed:
removed duplicate highlights toolbox

aferditamuriqi committed:
GitHub Actions / Build, lint, and test

Check warning on line 789 in src/modules/search/SearchModule.ts

Check warning on line 790 in src/modules/search/SearchModule.ts

aferditamuriqi committed:
GitHub Actions / Build, lint, and test

Check warning on line 451 in src/navigator/IFrameNavigator.ts

kristojorg committed:
Merge pull request #247 from d-i-t-a/dita/v2-build-system-fixes

Minor fixes.

aferditamuriqi reviewed (approved):
looks good :) once this is merged i'll add that fix for that strange bug i saw.
